src/engine/scheduler.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  EngineInterface,
+  SchedulerInterface,
+  FrameInterface
+} from './interfaces';
+import W from './window-proxy';
+
+import { default as Queue, DOMQueue} from './queues/queue';
+
+export class Frame implements FrameInterface {
+  constructor(
+    public timestamp: number,
+    public scrollTop: number,
+    public scrollLeft: number,
+    public width: number,
+    public height: number
+  ) {}
+}
+
+export function uuid() {
```

Comment:
We likely need to create benches for this if it's called a lot. Likely we can improve the perf here.